druid/src/widget/view_switcher.rs

```diff
@@ -75,10 +75,13 @@ impl<T: Data, U: PartialEq> Widget<T> for ViewSwitcher<T, U> {
 
     fn update(&mut self, ctx: &mut UpdateCtx, _old_data: &T, data: &T, env: &Env) {
         let child_id = (self.child_picker)(data, env);
-        if Some(&child_id) != self.active_child_id.as_ref() {
-            self.active_child = Some(WidgetPod::new((self.child_builder)(&child_id, data, env)));
-            self.active_child_id = Some(child_id);
-            ctx.children_changed();
+        if let Some(active_child_id) = &self.active_child_id {
+            if child_id.same(&active_child_id) {
+                self.active_child =
+                    Some(WidgetPod::new((self.child_builder)(&child_id, data, env)));
+                self.active_child_id = Some(child_id);
+                ctx.children_changed();
+            }
```

Comment:
There must be a more elegant a more elegant way to implement this control flow ...

Reply:
I think you're missing a negation in there - we're creating a new child if `child_id` is different from `active_child_id` (the new check here checks for sameness without negation)

Also, what happens here in the case where `active_child_id` is `None`? I think in that case we should be creating the appropriate child? This was handled by some-wrapping the non-optional `child_id` (returned from `child_picker`) in the old version. If the active one was none, we enter the change block.

Reply:
You were right about the negation. Fixed that.

As for the second question, I don't know. I only have a hazy understanding of how this component actually works. But it doesn't seem the existing code does this so perhaps that could be a separate PR?

Reply:
> Also, what happens here in the case where `active_child_id` is `None`? 

In that case, we would have a bug somewhere else. Because `Widget::lifecycle(LifeCycle::WidgetAdded)` must be the first event a widget receives and thus `active_child_id` can only be `None` if that rule gets violated somewhere.

Just doing
```rust
        if !child_id.same(self.active_child_id.as_ref().unwrap()) {
```
would be fine in my opinion.

Reply:
Then why is it `Option` at all?

Reply:
Because we have to initialize it lazily during `WidgetAdded`, and thus we have to assign some value (`None`) to it until we get there.

Reply:
Went with @Finnerale's suggestion here.